src/main/java/digital/slovensko/autogram/core/SigningJob.java

```diff
@@ -154,19 +156,28 @@ private DSSDocument signDocumentAsPAdeS(SigningKey key) {
     public static FileDocument createDSSFileDocumentFromFile(File file) {
         var fileDocument = new FileDocument(file);
 
-        if (isXDC(fileDocument.getMimeType()) || isXML(fileDocument.getMimeType()) && XDCValidator.isXDCContent(fileDocument))
-            fileDocument.setMimeType(AutogramMimeType.XML_DATACONTAINER);
+        if (fileDocument.getName().endsWith(".xdcf"))
+            fileDocument.setMimeType(XML_DATACONTAINER_WITH_CHARSET);
+
+        else if (isXDC(fileDocument.getMimeType()) || isXML(fileDocument.getMimeType()) && XDCValidator.isXDCContent(fileDocument))
+            fileDocument.setMimeType(AutogramMimeType.XML_DATACONTAINER_WITH_CHARSET);
+
+        else if (isTxt(fileDocument.getMimeType()))
+            fileDocument.setMimeType(AutogramMimeType.TEXT_WITH_CHARSET);
 
         return fileDocument;
     }
 
     private static SigningJob build(DSSDocument document, SigningParameters params, Responder responder) {
-        if (params.shouldCreateXdc()) {
-            var mimeType = document.getMimeType();
-            if (!isXDC(mimeType) && !isAsice(mimeType)) {
-                document = XDCBuilder.transform(params, document.getName(), EFormUtils.getXmlFromDocument(document));
-                document.setMimeType(AutogramMimeType.XML_DATACONTAINER);
-            }
+        if (params.shouldCreateXdc() && !isXDC(document.getMimeType()) && !isAsice(document.getMimeType()))
+            document = XDCBuilder.transform(params, document.getName(), EFormUtils.getXmlFromDocument(document));
+
+        if (isTxt(document.getMimeType()))
+            document.setMimeType(AutogramMimeType.TEXT_WITH_CHARSET);
+
+        if (isXDC(document.getMimeType())) {
+            document.setMimeType(AutogramMimeType.XML_DATACONTAINER_WITH_CHARSET);
```

Comment:
Dopady tohto neviem uplne posudit. Ale kedze je to standardny mimetype s charsetom, tak si nemame velmi co pokazit ze?

btw. ak sa da tak do pr napis aj preco sa ta zmena deje lebo nejako to neviem vycitat z toho kodu.

Reply:
Spravil som k tomu aj testovacie súbbory a celkovo máme testy na otváranie súborov, čiže by to malo byť fajn, a aj som to manuálne testoval.

Zmena sa deje preto, lebo Štefan kedysi písal, že toto je povinné tam uvádzať z vyhlášky ("charset=URF-8"). Okrem toho je vo vyhláške a už aj v IANA prípona .xdcf pre XDC súbory.